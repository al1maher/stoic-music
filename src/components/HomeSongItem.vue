<style scoped>
  .song-item-container {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  @media (max-width: 768px) {
    .song-item-container {
      max-width: 200px; /* Adjust the max-width as per your preference */
    }
  }
</style>

<template>
  <router-link
    :to="`/song/${song.docId}`"
    class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50"
    :id="`song-id-$`"
  >
    <div>
      <a href="#" class="font-bold block text-gray-600 song-item-container">{{ song.modified_name }}</a>
      <span class="text-gray-500 text-sm">{{ song.display_name }}</span>
    </div>

    <div class="text-gray-600 text-lg hover:text-yellow-500 transition duration-300 group">
      <router-link
        custom
        :to="{ name: 'song', params: { id: song.docId }, hash: '#comments' }"
        v-slot="{ navigate }"
      >
        <span class="comments" @click="navigate">
          <i
            class="fa fa-comments text-gray-600 group-hover:text-yellow-500 transition duration-300"
          ></i>
          {{ song.comment_count }}
        </span>
      </router-link>
    </div>
  </router-link>
</template>


<script>
export default {
  name: 'HomeSongItem',
  props: {
    song: {
      type: Object,
      required: true
    }
  },
}
</script>
