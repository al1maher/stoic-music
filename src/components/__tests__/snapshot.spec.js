import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import HomeSongItem from '@/components/HomeSongItem.vue'

describe('Snapshots HomeSongItem.vue', () => {
  test('renders correctly', () => {
    const song = {
      docId: 'abc',
      display_name: 'test',
      modified_name: 'test',
      comments_count: 0
    }

    const wrapper = shallowMount(HomeSongItem, {
      propsData: {
        song
      },
      global: {
        components: {
          'router-link': RouterLinkStub
        }
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
