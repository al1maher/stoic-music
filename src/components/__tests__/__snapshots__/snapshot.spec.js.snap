// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshots HomeSongItem.vue > renders correctly 1`] = `
<router-link-stub
  class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50"
  custom="false"
  id="song-id-abc"
  to="/song/abc"
/>
`;
